### PR TITLE
Update Linux Readme Build Steps

### PR DESCRIPTION
**Pull request by patrickdupuis**
## Purpose and Motivation

This PR replaces #4414 and adds to it by making obtaining the source code the first step. Thanks @enrike!

## Types of changes

- Documentation

## To-do list

- [ ] Code is tested
- [ ] All tests are passing
- [X] Updated documentation
- [X] This PR is ready for review

### CODE REVIEW

File: README_LINUX.md [hunk 1]
```diff
@@ -108,7 +108,17 @@ Unfortunately, the Qt installer does not allow you to deselect the multi-gigabyt
...
 --------
 
-### Step 1: Make a build directory
+### Step 1: Clone the repository
+
+SuperCollider is hosted on Github: https://github.com/SuperCollider/SuperCollider
+
+First, clone the repository with git:
+
+    git clone --recursive https://github.com/SuperCollider/SuperCollider.git
```
Comment (gusano) [inline README_LINUX.md]: I know I'm being picky but you shouldn't use uppercase characters in URLs

Comment (patrickdupuis) [reply README_LINUX.md]: I hadn't noticed that. Thanks!

**Comment by snappizz** [comment]
please remember to use [skip ci] :)

**Comment by dvzrv** [comment]
@patrickdupuis potentially one can also download a release tarball and build that, or is this meant to be solely about building from HEAD?

Also, would it be possible to add `cmake .. -L` (run in the `build` directory) as a step to retrieve potentially useful parameters to cmake for the project? I see only a few documented, but there are some (functioning) devendoring options (e.g. `-DSYSTEM_YAMLCPP=ON -DSYSTEM_BOOST=ON`), that are particularly useful on Linux distributions, as those libraries are usually packaged.

**Comment by patrickdupuis** [comment]
@dvzrv you make a good point about building from release tarballs. I'll add something to the first step.

As for the cmake flags, I think that could be for another PR. Would you have time to write something up?

File: README_LINUX.md [hunk 2]
```diff
@@ -108,7 +108,28 @@ Unfortunately, the Qt installer does not allow you to deselect the multi-gigabyt
...
+    git clone --recurse-submodules https://github.com/SuperCollider/SuperCollider.git
+
+The `--recurse-submodules` option will clone the repository's submodules which are needed to build SuperCollider. If you forget to clone them, you'll get error messages like the one listed below later on in the build process:
+
+  [SUBMODULE-NAME] is missing: please run `git submodule init && git
+  submodule update' from the toplevel of your git working tree
+
+The submodules can be cloned manually by navigating to the root your locally cloned SuperCollider repository and running the following command:
+
+  git submodule update --init
```
Comment (brianlheim) [inline README_LINUX.md]: I don't think any of this (from "If you forget to clone them" onward) needs to be restated; it's obvious from the error message. If you don't think it's clear enough, update that warning text in the cmake file.

Comment (patrickdupuis) [reply README_LINUX.md]: Thanks for reviewing @brianlheim. I removed the part mentioning the cmake error message. I like it much better now. @snappizz asked that I add this stuff during the last dev meeting.

Comment (snappizz) [reply README_LINUX.md]: yeah that nathan guy's a damn fool, don't listen to anything he says

Comment (patrickdupuis) [reply README_LINUX.md]: I realize that my comment came out harsher than I intended it to @snappizz. I simply wanted to let @brianlheim know that we discussed it during the last meeting. You ain't no damn fool :)

In any case, this bit of text is better now than it was. Thanks for the reviews!

File: README_LINUX.md [hunk 3]
```diff
@@ -108,7 +108,23 @@ Unfortunately, the Qt installer does not allow you to deselect the multi-gigabyt
...
+
+SuperCollider releases are available to download here: https://github.com/supercollider/supercollider/releases
+
+Cloning the repository can be done with the following command:
+
+    git clone --recurse-submodules https://github.com/SuperCollider/SuperCollider.git
+
+The `--recurse-submodules` option will clone the repository's submodules which are needed to build SuperCollider. The submodules can also be obtained by navigating to the root of your locally cloned SuperCollider repository and running the following command:
+
+  git submodule update --init
```
Comment (brianlheim) [inline README_LINUX.md]: i believe you also want --recursive here for ableton link. and please indent this with 4 spaces to match the rest of the added text

Comment (patrickdupuis) [reply README_LINUX.md]: thanks! I had no idea `--recursive` was needed here.

**Comment by wtkns** [comment]
fwiw, i just built this on ubuntu disco 19.04, and in addition to the provided apt-get commands, I also had to install libyaml-cpp-dev and asio-dev packages. dunno if those are worth adding to the instructions, but wanted to mention. thanks for your work on this!

**Pull request merged**